internal/controllers/device_data_controller_v2.go

```diff
@@ -0,0 +1,153 @@
+package controllers
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"strconv"
+
+	"github.com/DIMO-Network/devices-api/pkg/grpc"
+	"github.com/DIMO-Network/shared"
+	"github.com/gofiber/fiber/v2"
+	"github.com/pkg/errors"
+	"github.com/rs/zerolog"
+
+	"github.com/DIMO-Network/device-data-api/internal/config"
+	"github.com/DIMO-Network/device-data-api/internal/services"
+)
+
+type DeviceDataControllerV2 struct {
+	Settings  *config.Settings
+	log       *zerolog.Logger
+	deviceAPI services.DeviceAPIService
+	esService EsInterface
+}
+
+// NewDeviceDataControllerV2 constructor
+func NewDeviceDataControllerV2(settings *config.Settings, logger *zerolog.Logger, deviceAPIService services.DeviceAPIService, esService EsInterface) DeviceDataControllerV2 {
+	return DeviceDataControllerV2{
+		Settings:  settings,
+		log:       logger,
+		deviceAPI: deviceAPIService,
+		esService: esService,
+	}
+}
+
+// GetDailyDistance godoc
+// @Description  Get kilometers driven for a userDeviceID each day.
+// @Tags         device-data
+// @Produce      json
+// @Success      200 {object} controllers.DailyDistanceResp
+// @Failure      404 "no device found for user with provided parameters"
+// @Param        userDeviceID  path   string  true   "user device id"
+// @Param	     time_zone query string true "IANAS time zone id, e.g., America/Los_Angeles"
+// @Security     BearerAuth
+// @Router       /v2/vehicles/{tokenID}/analytics/daily-distance [get]
+func (d *DeviceDataControllerV2) GetDailyDistance(c *fiber.Ctx) error {
+	tz := c.Query("time_zone")
+	tkID := c.Params("tokenID")
+
+	device, err := d.getDeviceFromTokenID(c.Context(), tkID)
+	if err != nil {
+		return fiber.NewError(fiber.StatusBadRequest, "could find a device using provided tokenId!")
+	}
+
+	res, err := d.esService.GetTotalDailyDistanceDriven(c.Context(), tz, device.Id)
+	if err != nil {
+		return fmt.Errorf("failed to get total distance driven for device with token id %s - %w", tkID, err)
+	}
+
+	var ddr dailyDistanceElasticResult
+
+	err = json.Unmarshal(res, &ddr)
+	if err != nil {
+		return fmt.Errorf("failed to parse response %w", err)
+	}
+
+	buckets := ddr.Aggregations.Days.Buckets
+
+	days := make([]DailyDistanceDay, len(buckets))
+
+	for i, b := range buckets {
+		var dp *float64
+
+		if b.MaxOdom.Value != nil {
+			if shared.IsOdometerValid(*b.MaxOdom.Value) {
+				d := *b.MaxOdom.Value - *b.MinOdom.Value
+				dp = &d
+			}
+		}
+
+		day := DailyDistanceDay{
+			Date:     buckets[i].KeyAsString[:10],
```

Comment:
nit: I know this is copy pasted, but could you add a comment as to why we are stopping the first 10 characters?